packages/adapter-node/src/index.js

```diff
@@ -42,19 +42,25 @@ if (socket_activation) {
 	});
 }
 
-function shutdown() {
+/** @param {'SIGINT' | 'SIGTERM' | 'IDLE'} reason */
+function shutdown(reason) {
 	if (shutdown_timeout_id) return;
 
 	// @ts-expect-error this was added in 18.2.0 but is not reflected in the types
 	server.server.closeIdleConnections();
 
-	server.server.close(() => {
+	server.server.close((error) => {
+		if (error) return;
+
 		if (shutdown_timeout_id) {
 			shutdown_timeout_id = clearTimeout(shutdown_timeout_id);
 		}
 		if (idle_timeout_id) {
 			idle_timeout_id = clearTimeout(idle_timeout_id);
 		}
+
+		// @ts-expect-error Custom events cannot be typed
```

Comment:
just to match the style in the kit codebase
```suggestion
		// @ts-expect-error custom events cannot be typed
```